backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -2709,19 +2763,26 @@ public Response commitArtifactPart(CommitArtifactPart request) throws ModelDBExc
 
   private void deleteAllKeyValueEntities(
       Session session, String experimentRunId, String fieldType) {
-    Query query = session.createQuery(DELETE_ALL_KEY_VALUES_HQL);
-    query.setParameter(ModelDBConstants.EXPERIMENT_RUN_ID_STR, experimentRunId);
-    query.setParameter("field_type", fieldType);
-    query.executeUpdate();
+    session
+        .createNativeQuery(
+            "DELETE FROM keyvalue WHERE experiment_run_id = :experiment_run_id"
+                + " AND field_type = :field_type")
+        .setParameter("experiment_run_id", experimentRunId)
+        .setParameter("field_type", fieldType)
+        .executeUpdate();
```

Comment:
this is the same as old code, i do not understand what is the difference on both and purpose??

Reply:
Native queries bypass the hibernate session cache and statement cache.  This removes the extra processing required by hibernate to translate HQL into SQL, and prevents hibernate from caching the query or the results. 

Under high load, the extra processing time required by hibernate to translate and cache the query become a bottleneck.